manuscript/recipes/photoprism.md

```diff
@@ -0,0 +1,176 @@
+hero: Your own private google photos
+
+# Photoprism
+
+[Photoprism™](https://github.com/photoprism/photoprism) *" is a server-based application for browsing, organizing and sharing your personal photo collection. It makes use of the latest technologies to automatically tag and find pictures without getting in your way. Say goodbye to solutions that force you to upload your visual memories to the cloud."*
+
+
+![Photoprism Screenshot](../images/photoprism.png)
+
+## Ingredients
+
+1. [Docker swarm cluster](/ha-docker-swarm/design/) with [persistent shared storage](/ha-docker-swarm/shared-storage-ceph.md)
+2. [Traefik](/ha-docker-swarm/traefik_public) configured per design
+3. DNS entry for the hostname you intend to use, pointed to your [keepalived](ha-docker-swarm/keepalived/) IP
+
+## Preparation
+
+### Setup data locations
+
+First we need a folder to map the photoprism config file:
+```
+mkdir /var/data/photoprism/config
+```
+
+We will need a location to store photoprism thumbnails, as they can be recreated anytime (althought depending on your colleciton size it could take a while), we store them on a "non-backed" folder
+
+```
+mkdir /var/data/runtime/photoprism/cache
+```
+
+We will need to map three folders on our system / data:
+  1) originals - the folder where our original photo collection is stored (photoprism doesn't modify any original file, it only adds sidecars files).
+  2) import - the folder where photoprism will pick new photos to be added to the collection
+  3) export - the folder where photoprism will export photos.
+
+Photoprism has with its own running db, but if your collection is big (10K photos or more), the perfomance is best using an external db instance. We will use MariaDb, so we need the folders for running and backing the db:
+
+```
+mkdir /var/data/runtime/photoprism/db
+mkdir /var/data/photoprism/database-dump
+```
+
+### Prepare environment
+
+Create photoprism.env, and populate with the following variables. Change passwords
```

Comment:
When referencing filenames, use backticks for better markdown presentation. I.e., `photoprism.env`

Reply:
Added.